CRM/Queue/DAO/Queue.php

```diff
@@ -145,21 +181,89 @@ public static function &fields() {
           ],
           'add' => '5.47',
         ],
-        'is_autorun' => [
-          'name' => 'is_autorun',
-          'type' => CRM_Utils_Type::T_BOOLEAN,
-          'title' => ts('Enable Autorun'),
-          'description' => ts('Should the standard background attempt to autorun tasks in this queue?'),
-          'where' => 'civicrm_queue.is_autorun',
+        'runner' => [
+          'name' => 'runner',
+          'type' => CRM_Utils_Type::T_STRING,
+          'title' => ts('Runner'),
+          'description' => ts('Name of the task runner'),
+          'required' => FALSE,
+          'maxlength' => 64,
+          'size' => CRM_Utils_Type::BIG,
+          'where' => 'civicrm_queue.runner',
+          'table_name' => 'civicrm_queue',
+          'entity' => 'Queue',
+          'bao' => 'CRM_Queue_BAO_Queue',
+          'localizable' => 0,
+          'html' => [
+            'type' => 'Text',
+          ],
+          'add' => '5.48',
+        ],
+        'batch_limit' => [
+          'name' => 'batch_limit',
+          'type' => CRM_Utils_Type::T_INT,
+          'title' => ts('Batch Limit'),
+          'description' => ts('Maximum number of items in a batch. Tip: If you expand batch_limit, then also consider expanding lease_time.'),
+          'required' => TRUE,
+          'where' => 'civicrm_queue.batch_limit',
+          'default' => '1',
           'table_name' => 'civicrm_queue',
           'entity' => 'Queue',
           'bao' => 'CRM_Queue_BAO_Queue',
           'localizable' => 0,
           'html' => [
-            'type' => 'CheckBox',
-            'label' => ts("Auto Run"),
+            'type' => 'Text',
+          ],
+          'add' => '5.48',
+        ],
+        'lease_time' => [
+          'name' => 'lease_time',
+          'type' => CRM_Utils_Type::T_INT,
+          'title' => ts('Lease Time'),
+          'description' => ts('When claiming an item (or batch of items) for work, how long should the item(s) be reserved. (Seconds)'),
+          'required' => TRUE,
+          'where' => 'civicrm_queue.lease_time',
+          'default' => '3600',
+          'table_name' => 'civicrm_queue',
+          'entity' => 'Queue',
+          'bao' => 'CRM_Queue_BAO_Queue',
+          'localizable' => 0,
+          'html' => [
+            'type' => 'Text',
+          ],
+          'add' => '5.48',
+        ],
+        'retry_limit' => [
+          'name' => 'retry_limit',
+          'type' => CRM_Utils_Type::T_INT,
+          'title' => ts('Retry Limit'),
+          'description' => ts('Number of permitted retries. Decreases with each retry. Zero (0) to disable. Null for system default.'),
```

Comment:
Possibly out of scope of the PR - but what is the system default - is it just hard coded somewhere else? In which case this field could just have it as a default....

Reply:
Re-reading the current draft of #22762... it looks like there is no system default.

* In some previous/intermediate draft, there had been a system default (distinct from the SQL default). Removing it made the code simpler. Now it should just abide the SQL default.
* The tasks in a queue should be amenable to retrying (eg idempotent code or transactional code or defensive code). I suspect it's easy for many tasks to be amenable, but it's not automatic/universal.
* There's a difference between "default" as "typical value" versus "fallback/implied value":
    * I think `retry_limit=2` works as a "typical value". We could use that in examples/dev-docs.
    * I think `retry_limit=0` is sensible as the "fallback/implied value". If the developer isn't actively thinking about retries, then it's safer to assume that retries won't work.
    * (*The `get` API has a `limit` which seems like a good analog. API Explorer encourages `limit=25` as a "typical value", but the "fallback/implied value" is `limit=null`.)

I'll push up some changes to make this is a straight default of `retry_limit=0`.

Reply:
Updated to simpler default (`retry_limit=0`).